Source/options.js

```diff
@@ -453,5 +453,39 @@ $(function() {
     save('ngramMax', 'Maximum word sequence length saved');
   }
 
+  // Listener to update blacklist from context menu
+  chrome.runtime.onMessage.addListener(function(request, sender, sendResponse) {
```

Comment:
You are right. Duplication is not good. I will merge it for now then. Could you add the remark I raised, as well as your reply, as a "ToDo" comment in the code? This will remind us to think about a better solution in the future.